### PR TITLE
Allow FlxBullet offset to count for parent's angle

### PR DESCRIPTION
**Pull request by alihassan0**
to easily test just use this project [flxWeaponTest.zip](https://github.com/HaxeFlixel/flixel-addons/files/278162/flxWeaponTest.zip)

### CODE REVIEW

File: flixel/addons/weapon/FlxWeapon.hx [hunk 1]
```diff
@@ -479,7 +513,7 @@ class FlxTypedWeapon<TBullet:FlxBullet>
 
 enum FlxWeaponFireFrom
 {
-	PARENT(parent:FlxSprite, offset:FlxBounds<FlxPoint>);
+	PARENT(parent:FlxSprite, offset:FlxBounds<FlxPoint>, ?useParentAngle:Bool);
```
Comment (Gama11) [inline flixel/addons/weapon/FlxWeapon.hx]: Can you replace `?useParentAngle:Bool` with `useParentAngle:Bool = false`? There's no good reason to turn this into a `Null<Bool>` (see http://stackoverflow.com/questions/31914614/is-question-mark-operator-in-haxe-really-necessary/31916466#31916466).

Comment (alihassan0) [inline flixel/addons/weapon/FlxWeapon.hx]: Done :+1:

**Comment by Gama11** [comment]
Thanks! The test project you provided was quite helpful for reviewing the changes. :+1:

**Comment by Gama11** [comment]
Hm, nevermind about my earlier comment, I could've sworn that compiled when I tried it - seems like Haxe simply doesn't allow default values for enum constructors like it does with functions.s

**Pull request merged**